cms/djangoapps/contentstore/api/tests/test_views.py

```diff
@@ -0,0 +1,142 @@
+            f.write('<course></course>')
+
+        cls.good_tar = os.path.join(cls.content_dir, "good.tar.gz")
+        with tarfile.open(cls.good_tar, "w:gz") as gtar:
+            gtar.add(good_dir)
+
+        cls.namespaced_url = 'courses_api:course_import'
+
+    def setUp(self):
+        super(CourseImportViewTest, self).setUp()
+
+    def get_url(self, course_id=None):
+        """
+        Helper function to create the url
+        """
+        if(course_id is None):
+            course_id = self.course_key
+        base_url = reverse(
+            self.namespaced_url,
+            kwargs={
+                'course_id': course_id
+            }
+        )
+        query_string = ''
+        return base_url + query_string
+
+    def test_anonymous_import_fails(self):
+        """
+        Test that an anonymous user cannot access the API and an error is received.
+        """
+        with open(self.good_tar, 'rb') as fp:
+            resp = self.client.post(self.get_url(), {'course_data': fp}, format='multipart')
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_student_import_fails(self):
+        """
+        Test that an student user cannot access the API and an error is received.
+        """
+        self.client.login(username=self.student.username, password=self.password)
+        with open(self.good_tar, 'rb') as fp:
+            resp = self.client.post(self.get_url(), {'course_data': fp}, format='multipart')
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_staff_with_access_import_succeeds(self):
+        """
+        Test that an staff user can access the API and successfully upload a course
+        """
+        self.client.login(username=self.staff.username, password=self.password)
+        with open(self.good_tar, 'rb') as fp:
+            resp = self.client.post(self.get_url(), {'course_data': fp}, format='multipart')
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_staff_has_no_access_import_fails(self):
+        """
+        Test that an staff user can't access another course via the API
+        """
+        self.client.login(username=self.staff.username, password=self.password)
+        with open(self.good_tar, 'rb') as fp:
+            resp = self.client.post(self.get_url(self.restricted_course_key), {'course_data': fp}, format='multipart')
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_student_get_status_fails(self):
+        """
+        Test that an student user cannot access the API and an error is received.
+        """
+        self.client.login(username=self.student.username, password=self.password)
+        resp = self.client.get(self.get_url(), {'task_id': '1234'})
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_staff_get_status_succeeds(self):
```

Comment:
Do you think it's worth adding anonymous and no-access staff user test cases for completeness?